File: src/frontend/cadastros/lib/app/data/model/pessoa_endereco_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:cadastros/app/infra/infra_imports.dart';
import 'package:cadastros/app/data/domain/domain_imports.dart';

class PessoaEnderecoModel {
	int? id;
	int? idPessoa;
	String? logradouro;
	String? numero;
	String? complemento;
	String? bairro;
	String? cidade;
	String? uf;
	String? cep;
	int? municipioIbge;
	String? principal;
	String? entrega;
	String? cobranca;
	String? correspondencia;

	PessoaEnderecoModel({
		this.id,
		this.idPessoa,
		this.logradouro,
		this.numero,
		this.complemento,
		this.bairro,
		this.cidade,
		this.uf,
		this.cep,
		this.municipioIbge,
		this.principal,
		this.entrega,
		this.cobranca,
		this.correspondencia,
	});

	static List<String> dbColumns = <String>[
		'id',
		'logradouro',
		'numero',
		'complemento',
		'bairro',
		'cidade',
		'uf',
		'cep',
		'municipio_ibge',
		'principal',
		'entrega',
		'cobranca',
		'correspondencia',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Logradouro',
		'Numero',
		'Complemento',
		'Bairro',
		'Cidade',
		'Uf',
		'Cep',
		'Municipio Ibge',
		'Principal',
		'Entrega',
		'Cobranca',
		'Correspondencia',
	];

	PessoaEnderecoModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idPessoa = jsonData['idPessoa'];
		logradouro = jsonData['logradouro'];
		numero = jsonData['numero'];
		complemento = jsonData['complemento'];
		bairro = jsonData['bairro'];
		cidade = jsonData['cidade'];
		uf = PessoaEnderecoDomain.getUf(jsonData['uf']);
		cep = jsonData['cep'];
		municipioIbge = jsonData['municipioIbge'];
		principal = PessoaEnderecoDomain.getPrincipal(jsonData['principal']);
		entrega = PessoaEnderecoDomain.getEntrega(jsonData['entrega']);
		cobranca = PessoaEnderecoDomain.getCobranca(jsonData['cobranca']);
		correspondencia = PessoaEnderecoDomain.getCorrespondencia(jsonData['correspondencia']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idPessoa'] = idPessoa != 0 ? idPessoa : null;
		jsonData['logradouro'] = logradouro;
		jsonData['numero'] = numero;
		jsonData['complemento'] = complemento;
		jsonData['bairro'] = bairro;
		jsonData['cidade'] = cidade;
		jsonData['uf'] = PessoaEnderecoDomain.setUf(uf);
		jsonData['cep'] = Util.removeMask(cep);
		jsonData['municipioIbge'] = municipioIbge;
		jsonData['principal'] = PessoaEnderecoDomain.setPrincipal(principal);
		jsonData['entrega'] = PessoaEnderecoDomain.setEntrega(entrega);
		jsonData['cobranca'] = PessoaEnderecoDomain.setCobranca(cobranca);
		jsonData['correspondencia'] = PessoaEnderecoDomain.setCorrespondencia(correspondencia);
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idPessoa = plutoRow.cells['idPessoa']?.value;
		logradouro = plutoRow.cells['logradouro']?.value;
		numero = plutoRow.cells['numero']?.value;
		complemento = plutoRow.cells['complemento']?.value;
		bairro = plutoRow.cells['bairro']?.value;
		cidade = plutoRow.cells['cidade']?.value;
		uf = plutoRow.cells['uf']?.value != '' ? plutoRow.cells['uf']?.value : 'AC';
		cep = plutoRow.cells['cep']?.value;
		municipioIbge = plutoRow.cells['municipioIbge']?.value;
		principal = plutoRow.cells['principal']?.value != '' ? plutoRow.cells['principal']?.value : 'Sim';
		entrega = plutoRow.cells['entrega']?.value != '' ? plutoRow.cells['entrega']?.value : 'Sim';
		cobranca = plutoRow.cells['cobranca']?.value != '' ? plutoRow.cells['cobranca']?.value : 'Sim';
		correspondencia = plutoRow.cells['correspondencia']?.value != '' ? plutoRow.cells['correspondencia']?.value : 'Sim';
	}	

	PessoaEnderecoModel clone() {
		return PessoaEnderecoModel(
			id: id,
			idPessoa: idPessoa,
			logradouro: logradouro,
			numero: numero,
			complemento: complemento,
			bairro: bairro,
			cidade: cidade,
			uf: uf,
			cep: cep,
			municipioIbge: municipioIbge,
			principal: principal,
			entrega: entrega,
			cobranca: cobranca,
			correspondencia: correspondencia,
		);			
	}

	
}

File: src/backend/cadastros/src/model/pessoa_endereco_model.py
from src import db


class PessoaEnderecoModel(db.Model):
    __tablename__ = 'pessoa_endereco'

    id = db.Column(db.Integer, primary_key=True)
    logradouro = db.Column(db.String(100))
    numero = db.Column(db.String(10))
    complemento = db.Column(db.String(100))
    bairro = db.Column(db.String(100))
    cidade = db.Column(db.String(100))
    uf = db.Column(db.String(2))
    cep = db.Column(db.String(8))
    municipio_ibge = db.Column(db.Integer)
    principal = db.Column(db.String(1))
    entrega = db.Column(db.String(1))
    cobranca = db.Column(db.String(1))
    correspondencia = db.Column(db.String(1))
    id_pessoa = db.Column(db.Integer, db.ForeignKey('pessoa.id'))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_pessoa = data.get('idPessoa')
        self.logradouro = data.get('logradouro')
        self.numero = data.get('numero')
        self.complemento = data.get('complemento')
        self.bairro = data.get('bairro')
        self.cidade = data.get('cidade')
        self.uf = data.get('uf')
        self.cep = data.get('cep')
        self.municipio_ibge = data.get('municipioIbge')
        self.principal = data.get('principal')
        self.entrega = data.get('entrega')
        self.cobranca = data.get('cobranca')
        self.correspondencia = data.get('correspondencia')

    def serialize(self):
        return {
            'id': self.id,
            'idPessoa': self.id_pessoa,
            'logradouro': self.logradouro,
            'numero': self.numero,
            'complemento': self.complemento,
            'bairro': self.bairro,
            'cidade': self.cidade,
            'uf': self.uf,
            'cep': self.cep,
            'municipioIbge': self.municipio_ibge,
            'principal': self.principal,
            'entrega': self.entrega,
            'cobranca': self.cobranca,
            'correspondencia': self.correspondencia,
        }